oeps/oep-0006.rst

```diff
@@ -0,0 +1,410 @@
+    intended to avoid the temptation to use this type of XBlock field for
+    logging or message passing. Such uses could cause data loss or performance
+    concerns, because the XBlock API does not support appends nor locking for
+    field data.
+  * Example: Enable or disable notifications about new discussion posts in this
+    course
+  * Example: Video playback speed for this course
+  * Example: An XBlock in which students "accumulate building blocks over the
+    course of the semester" (`PLAT-325`_)
+
+.. _PLAT-325: https://openedx.atlassian.net/browse/PLAT-325
+
+New XBlock view: context_settings_view
+--------------------------------------
+
+An XBlock which uses ``Scope.context_settings`` fields will need a way for
+content authors to edit the data in those fields. (e.g. when creating a course,
+the course author may wish to change course-wide settings for the discussion
+XBlock used throughout that course.)
+
+A new XBlock view, ``context_settings_view`` shall be used for this purpose.
+Just like the existing `studio_view` defined by edX Studio, this new view will
+allow an XBlock to return an arbitrary HTML Fragment which contains a form or
+other UI for editing its ``Scope.context_settings`` fields.
+
+It is expected (but not required) that runtimes like edX Studio which want to
+support editing of context-scoped fields will define a list of XBlocks that are
+enabled within a particular context/course, and will display a list of XBlocks
+that define the ``context_settings_view``. Clicking on the name of that XBlock
+would display the HTML fragment and allow the author to edit the context-scoped
+fields.
+
+.. _studio_view: http://edx.readthedocs.io/projects/xblock-tutorial/en/latest/edx_platform/edx_studio.html
+
+Context-scoped fields in OLX
+----------------------------
+
+When serializing a course with ``Scope.context_settings`` fields to Open
+Learning XML (OLX), any context-scoped fields should be described within a new
+``<xblock-settings>`` XML element that is a child of the ``<course>`` element.
+This would typically be found in ``course/course.xml``. Each XBlock class that
+has ``Scope.context_settings`` field data should be represented as a child
+element of ``<xblock-settings>``, with the name of the element matching the
+XBlock's entry point name.
+
+An example follows:
+
+.. code-block:: xml
+
+   <course display_name="Example Course" language="en" advanced_modules='["drag-and-drop-v2", "lti_consumer", "xblock-dalite"]'>
+     <chapter url_name="6915ee3dd6ab403d8c05c0ea3180a0ee"/>
+     <wiki slug="A.B.C"/>
+     <xblock-settings>
+       <discussion allow_anonymous_discussion_posts="true" discussion_blackout_dates='[["2015-09-15", "2015-09-21"], ["2015-10-01", "2015-10-08"]]' />
+       <lti_consumer lti_passports='["myapp:key1:secret1", "otherapp:key2:secret2"]' />
+       <xblock-dalite dalite_lti_passports='["dalite-ng-demo:https://dalite-ng-demo.example:key:secret"]' />
+     </xblock-settings>
+   </course>
+
```

Comment:
@cpennington: You could, but I think it would be more confusing, particularly if we ever mixed them up in the same document.

@bradenmacdonald: While it's true that nothing stops people from using an arbitrary tag name, I do think we can reasonably enforce a convention, even if the actual mapping happens in the form of explicit registry in entry points. I think that in general, the namespaces should be reserved on an XBlock basis -- so that it's main job is to disambiguate two "video" tags from two XBlocks, rather than two scoped tags from the same XBlock.

In any case, I apologize for stirring up this particular pot. I thought that OLX representation could be straightforward and simple, but I don't want to drag us into a full definition of OLX representations across different scopes as part of this OEP. Do you folks think it's reasonable to accept this OEP on the new scope, and punt on the serialization aspect as a separate follow-on step?

Thank you.